sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -192,6 +192,17 @@ case class InsertIntoHiveTable(
     if (partition.nonEmpty) {
       if (numDynamicPartitions > 0) {
         if (overwrite && table.tableType == CatalogTableType.EXTERNAL) {
+          val numWrittenParts = writtenParts.size
+          val maxDynamicPartitionsKey = "hive.exec.max.dynamic.partitions"
+          val maxDynamicPartitions = hadoopConf.getInt(maxDynamicPartitionsKey, 1000)
+          if (numWrittenParts > maxDynamicPartitions) {
+            val maxDynamicPartitionsErrMsg =
+              s"Number of dynamic partitions created is $numWrittenParts" +
+                s", which is more than $maxDynamicPartitions" +
+                s". To solve this try to set $maxDynamicPartitionsKey" +
+                s" to at least $numWrittenParts."
```

Comment:
Do you think we set `hive.exec.max.dynamic.partitions` automatically from Spark side in this case?

Reply:
It is possible to automatically adjust the number of `hive.exec.max.dynamic.partitions`. 
However, if it is automatically adjusted, many partitions may be created accidentally, and this parameter is meaningless. 

https://github.com/apache/hive/blob/135629b8d6b538fed092641537034a9fbc59c7a0/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java#L1857-L1864